src/MSIdentityScaffolding/Microsoft.DotNet.MSIdentity/CodeReaderWriter/ProjectModifier.cs

```diff
@@ -0,0 +1,209 @@
+                        }
+                        codeModifierConfigs.Add(projectDescription);
+                    }
+                }
+            }
+
+            var codeModifierConfig = codeModifierConfigs
+                .Where(x => x.Identifier != null &&
+                       x.Identifier.Equals(_toolOptions.ProjectTypeIdentifier, StringComparison.OrdinalIgnoreCase))
+                        .FirstOrDefault();
+
+            // CodeModifierConfig, .csproj path cannot be null
+            if (codeModifierConfig != null &&
+                codeModifierConfig.Files != null &&
+                codeModifierConfig.Files.Any())
+            {
+                return codeModifierConfig;
+            }
+            return null;
+        }
+
+        private CodeModifierConfig? ReadCodeModifierConfigFromFileContent(byte[] fileContent)
+        {
+            string jsonText = Encoding.UTF8.GetString(fileContent);
+            return JsonSerializer.Deserialize<CodeModifierConfig>(jsonText);
+        }
+
+        internal string GetClassName(string? className)
+        {
+            string formattedClassName = string.Empty;
+            if (!string.IsNullOrEmpty(className))
+            {
+                string[] blocks = className.Split(".cs");
+                if (blocks.Length > 1)
+                {
+                    return blocks[0];
+                }
+            }
+            return formattedClassName;
+        }
+
+        //Get Startup class name from CreateHostBuilder in Program.cs. If Program.cs is not being used, method
+        //will bail out.
+        internal async Task<string?> GetStarupClass(CodeAnalysis.Project project)
+        {
+            var programFilePath = Directory.EnumerateFiles(_toolOptions.ProjectPath, "Program.cs").FirstOrDefault();
+            if (!string.IsNullOrEmpty(programFilePath))
+            {
+                var programDoc = project.Documents.Where(d => d.Name.Equals(programFilePath)).FirstOrDefault();
+                var startupClassName = await GetStartupClassName(programDoc);
+                string className = startupClassName;
+                var startupFilePath = string.Empty;
+                if (!string.IsNullOrEmpty(startupClassName))
+                {
+                    return string.Concat(startupClassName, ".cs");
+                }
+            }
+            return string.Empty;
+        }
+
+        internal async Task<string> GetStartupClassName(Document? programDoc)
+        {
+            if (programDoc != null && await programDoc.GetSyntaxRootAsync() is CompilationUnitSyntax root)
+            {
+                var namespaceNode = root.Members.OfType<NamespaceDeclarationSyntax>()?.FirstOrDefault();
+                var programClassNode = namespaceNode?.DescendantNodes().Where(node => node is ClassDeclarationSyntax cds
+                && cds.Identifier.ValueText.Contains("Program")).First();
```

Comment:
>&& cds.Identifier.ValueText.Contains("Program")).First(); [](http://example.com/codeflow?start=16&length=57)

format - indent it for readability

Reply:
fixed and moved https://github.com/dotnet/Scaffolding/pull/1593

Reply:
fixed and moved https://github.com/dotnet/Scaffolding/pull/1593